client/src/store/index.js

```diff
@@ -634,88 +628,85 @@ mutations: {
-      localStorage.removeItem('COMPINFO')
-
-      localStorage.removeItem('kctoken')
+      localStorage.removeItem('KEYCLOAK_REFRESH')
       localStorage.removeItem('KEYCLOAK_TOKEN')
+      localStorage.removeItem('KEYCLOAK_EXPIRES')
       localStorage.removeItem('AUTHORIZED')
       localStorage.removeItem('USERNAME')
+      localStorage.removeItem('USER_ROLE')
 
     },
 
     loadSetUp({dispatch}){
+
+        // clear values from local storeage
+        dispatch('logout')
+
         //Read Configuration.json File
         readJFile('static/config/configuration.json', function (myArray) {
         axios.defaults.baseURL = myArray[0]['URL']
         console.log("Setting axios.baseURL to: " + axios.defaults.baseURL)
 
         //load UI dropdowns from json files and database tables
         dispatch('loadDropdowns');
+
       })
     },
 
     tryAutoLogin ({commit}) {
     },
 
-    storeUser ({commit, state}, userData) {
-      if (!state.idToken) {
-        return
+    checkToken({dispatch, state}){
+      // checks if keycloak object exists - if not then state is unstable, force logout
+      if(state.myKeycloak==null){
+        console.log('myKeycloak is null')
+        //TODO - reset everything and force login???
+        dispatch('logout')
+       return
       }
-      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
-        .then(res => console.log(res))
-        .catch(error => console.log(error))
-    },
+      // checks if keycloak object has tokenParsed yet, if not then just return as this only happens at login
+      if(state.myKeycloak.tokenParsed==null){ return }
 
-    fetchUser ({commit, state}) {
-      if (!state.idToken) {
-        return
+      var expiresIn = state.myKeycloak.tokenParsed['exp'] - Math.ceil(new Date().getTime() / 1000)
+
+      if(expiresIn < 1700 && expiresIn > 0) {
+        console.log('Token expires in ' + expiresIn + 'seconds, updating')
+        dispatch('updateToken')
+      }else if(expiresIn < 0) {
+        dispatch('logout')
       }
-      globalAxios.get('/users.json' + '?auth=' + state.idToken)
-        .then(res => {
-          console.log(res)
-          const data = res.data
-          const users = []
-          for (let key in data) {
-            const user = data[key]
-            user.id = key
-            users.push(user)
-          }
-          console.log(users)
-          commit('storeUser', users[0])
-        })
-        .catch(error => console.log(error))
+    },
+
+    updateToken({commit, state}){
+      const vm = this;
+      state.myKeycloak.updateToken(-1).success(function (refreshed) {
+        if (refreshed) {
+          localStorage.setItem('KEYCLOAK_TOKEN', state.myKeycloak.token);
+          localStorage.setItem('KEYCLOAK_REFRESH', state.myKeycloak.refreshToken);
+          localStorage.setItem('KEYCLOAK_EXPIRES', state.myKeycloak.tokenParsed.exp * 1000);
+          commit('authUser', state.myKeycloak.token)
```

Comment:
this references authUser mutation, but it's been deleted. Should it have been deleted? Or is it actually no longer needed and this call to commit('authUser'...) not needed?